### PR TITLE
Adds Permissions, Roles and Snowflake IDS

### PR DESCRIPTION
**Pull request by mohamed040406**
## Summary

Added Permissions and Roles Models and the SQL functions for `Snowflake IDS` and checking for global permissions

## Checklist

- [x] If endpoints were changed then they have been documented and tested.
    - [ ] I have updated the docmentation to reflect the changes.
    - [ ] I have updated the tests to support the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new endpoint or parameter).
- [ ] This PR is a breaking change (e.g. endpoint or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: api/models/permission.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        permissions = [
+            {
+                "name": "Administrator",
+                "value": cls.ADMINISTRATOR,
+                "public": False,
+                "help_text": "Users with this permission will have every permission"
+                " and will also bypass all channel specific permissions or restrictions.",
+            },
+            {
+                "name": "Manage Global Roles",
```
Comment (SylteA) [inline api/models/permission.py]: No need for "global" keyword

File: api/models/permission.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    async def create_all(cls, verbose: bool = False) -> List["Permission"]:
+        """Create all permissions required for the website."""
+
+        permissions = [
+            {
+                "name": "Administrator",
+                "value": cls.ADMINISTRATOR,
+                "public": False,
+                "help_text": "Users with this permission will have every permission"
+                " and will also bypass all channel specific permissions or restrictions.",
```
Comment (SylteA) [inline api/models/permission.py]: "channel specific permissions"?..

File: api/models/permission.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            },
+            {
+                "name": "Manage Global Roles",
+                "value": cls.MANAGE_ROLES,
+                "public": False,
+                "help_text": "Allows management and editing of global roles.",
+            },
+            {
+                "name": "Create Timathon",
```
Comment (SylteA) [inline api/models/permission.py]: Dont make permissions timathon specific, you also have weekly challenges.

**Review by SylteA — CHANGES_REQUESTED** [review]
When you finish the requested changes
Either finish role related endpoints or push this to its own branch so someone can finish it.

**Comment by SylteA** [comment]
Needs tests & documentation to be finished before merging.

**Comment by mohamed040406** [comment]
nvm

**Pull request closed without merging**